scripts/compress_images_test.py

```diff
@@ -0,0 +1,250 @@
+        img = Image.new('RGB', (100, 100), color='red')
+        img.save(path, format=Img_format)
+
+    def test_get_compression_type_zip_extension(self) -> None:
+        """Test get_compression_type returns 'Zip' for PNG."""
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        compression_type = compressor.get_compression_type('.png')
+        self.assertEqual(compression_type, 'Zip')
+
+    def test_get_compression_type_lzw_extension(self) -> None:
+        """Test get_compression_type returns 'LZW' for JPG."""
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        compression_type = compressor.get_compression_type('.jpg')
+        self.assertEqual(compression_type, 'LZW')
+
+    def test_find_compressible_images(self) -> None:
+        """Test find_compressible_images method."""
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        result_images: List[CompressedImageInfo] = (
+            compressor.find_compressible_images()
+        )
+
+        self.assertEqual(len(result_images), 1)
+
+        for image in result_images:
+            self.assertIn(image['path'].suffix, ['.png', '.jpg', '.webp'])
+            self.assertTrue(image['original_size'] > image['new_size'])
+
+    def test_compress_images(self) -> None:
+        """Test image compression process."""
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        result_images: List[CompressedImageInfo] = (
+            compressor.find_compressible_images()
+        )
+
+        self.assertGreaterEqual(len(result_images), 1)
+        compressor.compress_images(result_images)
+        for image in result_images:
+            original_path = image['path']
+            original_size = image['original_size']
+
+            rel_path = os.path.relpath(original_path, self.temp_dir)
+            output_path = pathlib.Path(self.output_dir) / rel_path
+
+            self.assertTrue(output_path.exists())
+            compressed_size = output_path.stat().st_size
+            self.assertLess(compressed_size, original_size)
+
+    def test_compress_images_directory_removal_error(self) -> None:
+        """Test error handling when removing output directory fails."""
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        os.makedirs(self.output_dir, exist_ok=True)
+        mock_rmtree = mock.MagicMock(ignore_errors=OSError('Permission denied'))
+        mock_log_error = mock.MagicMock()
+        mock_subprocess_result = mock.Mock(spec_set=['returncode', 'stderr'])
+        mock_subprocess_result.returncode = 1
+        mock_subprocess_result.stderr = 'Permission denied error'
+        mock_subprocess_run = mock.Mock(return_value=mock_subprocess_result)
```

Comment:
It looks like you're creating a permission denied error during conversion, not during directory removal like the test name suggests